interface/modules/custom_modules/oe-module-faxsms/library/api_onetime.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * Portal OneTime for API
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2023 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General public License 3
+ */
+
+require_once(__DIR__ . "/../../../../globals.php");
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Services\PatientPortalService;
+
+if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"], 'contact-form')) {
+    CsrfUtils::csrfNotVerified();
+}
+
+if (isset($_REQUEST['sendOneTime'])) {
+    $rtn = doOnetimeRequest();
+}
+
+function doOnetimeRequest()
+{
+    $service = new PatientPortalService();
+    $details = json_decode($service->getRequest('details'), true);
+    $content = $service->getRequest('comments');
+    $data = [
+        'pid' => $details['pid'] ?? 0,
+        'onetime_period' => $details['onetime_period'] ?? 'PT60M',
+        'notification_template_name' => $details['notification_template_name'] ?? '',
+        'document_id' => $details['id'] ?? 0,
+        'audit_id' => $details['audit_id'] ?? 0,
+        'document_name' => $details['template_name'] ?? '',
+        'notification_method' => $service->getRequest('notification_method', 'both'),
+        'phone' => $details['phone'] ?? '',
+        'email' => $details['email'] ?? '',
```

Comment:
does this allow sending to a separate email address or phone number that is not set in patient_data for a pid?

Reply:
No once I finish setting up new patient portal service the event data and the database data will be validated against each other. Depending on context and service type IE fax SMS email will determine whether those fields are disabled or not